### PR TITLE
feat: uuid v4 validation matcher

### PR DESCRIPTION
**Pull request by phos-web**
added validation matcher which checks if value is a UUID v4

### CODE REVIEW

File: core/citrus-base/src/main/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcher.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+            if (uuid.version() != 4) {
+                extracted(fieldName, value);
+            }
+        } catch (IllegalArgumentException e) {
+            extracted(fieldName, value);
+        }
+    }
+
+    private void extracted(String fieldName, String value) {
```
Comment (bbortt) [inline core/citrus-base/src/main/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcher.java:43]: this name is auto-generated, please adapt it.

```suggestion
    private void throwValidationException(String fieldName, String value) {
```

File: core/citrus-base/src/test/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcherTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        assertThatThrownBy(() ->
+        matcher.validate("field", "34d68c48-1455-43ac-a0d6-b7b894c7a7d2uuidtoolong", null, context))
+                .isInstanceOf(ValidationException.class)
+                .hasMessage("UuidV4ValidationMatcher failed for field 'field'. Received value '34d68c48-1455-43ac-a0d6-b7b894c7a7d2uuidtoolong' is not a uuid v4.");
+    }
+
+    @Test
+    public void testWrongUuidVersion() {
+        assertThatThrownBy(() ->
+        matcher.validate("field", "34be571d-7180-3bcf-bbda-20cffbfae9ed", null, context))
```
Comment (bbortt) [inline core/citrus-base/src/test/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcherTest.java:51]: I think this file needs an auto-formatting ("ctrl + alt + l" in Intellij) - this lines in the `assertThatThrownBy`-block are not properly formatted.

**Review by bbortt — CHANGES_REQUESTED** [review]
thank you for your contribution. I've only left two small remarks.

File: core/citrus-base/src/main/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcher.java (hunk 1 above)
Comment (phos-web) [reply core/citrus-base/src/main/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcher.java]: changed method name

File: core/citrus-base/src/test/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcherTest.java (hunk 2 above)
Comment (phos-web) [reply core/citrus-base/src/test/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcherTest.java]: did auto-formatting. "matcher.validate..." is now indented correctly.

File: core/citrus-base/src/test/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcherTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+package org.citrusframework.validation.matcher.core;
+
+import org.citrusframework.UnitTestSupport;
+import org.citrusframework.exceptions.ValidationException;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Fail.fail;
+
```
Comment (bbortt) [inline core/citrus-base/src/test/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcherTest.java:25]:
```suggestion
```

File: core/citrus-base/src/main/java/org/citrusframework/validation/matcher/DefaultValidationMatcherLibrary.java [hunk 4]
```diff
@@ -16,28 +16,7 @@
...
-import org.citrusframework.validation.matcher.core.MatchesValidationMatcher;
-import org.citrusframework.validation.matcher.core.NotEmptyValidationMatcher;
-import org.citrusframework.validation.matcher.core.NotNullValidationMatcher;
-import org.citrusframework.validation.matcher.core.NullValidationMatcher;
-import org.citrusframework.validation.matcher.core.StartsWithValidationMatcher;
-import org.citrusframework.validation.matcher.core.StringLengthValidationMatcher;
-import org.citrusframework.validation.matcher.core.TrimAllWhitespacesValidationMatcher;
-import org.citrusframework.validation.matcher.core.TrimValidationMatcher;
-import org.citrusframework.validation.matcher.core.WeekdayValidationMatcher;
+import org.citrusframework.validation.matcher.core.*;
```
Comment (bbortt) [inline core/citrus-base/src/main/java/org/citrusframework/validation/matcher/DefaultValidationMatcherLibrary.java:19]: please try reverting this. in IntelliJ, you can do this by settings:

image

**Review by bbortt — CHANGES_REQUESTED** [review]

**Review by bbortt — APPROVED** [review]
/approve

**Pull request merged**